File: packages/aws-durable-execution-conformance-tests-otel/examples/python/src/otel_2_wait_resume.py
# SPDX-FileCopyrightText: 2026-present Amazon.com, Inc. or its affiliates.
#
# SPDX-License-Identifier: Apache-2.0
"""Wait and resume scenario for OTel requirement otel-2."""

from __future__ import annotations

from typing import Any

from aws_durable_execution_sdk_python import (
    DurableContext,
    StepContext,
    durable_execution,
    durable_step,
)
from aws_durable_execution_sdk_python.config import Duration
from aws_durable_execution_sdk_python_otel import OtelPlugin
from common import require_scenario


@durable_step
def complete_after_resume(_step_context: StepContext) -> str:
    return "resumed"


@durable_execution(plugins=[OtelPlugin()])
def handler(event: dict[str, Any], context: DurableContext) -> str:
    require_scenario(event, "wait-resume")
    context.wait(Duration.from_seconds(1), name="otel-wait")
    return context.step(complete_after_resume(), name="otel-after-resume")

File: packages/aws-durable-execution-conformance-tests-otel/examples/java/src/main/java/software/amazon/lambda/durable/conformance/otel/Otel2WaitResume.java
// SPDX-FileCopyrightText: 2026-present Amazon.com, Inc. or its affiliates.
//
// SPDX-License-Identifier: Apache-2.0
package software.amazon.lambda.durable.conformance.otel;

import java.time.Duration;
import java.util.Map;
import software.amazon.lambda.durable.DurableContext;

/** Wait-and-resume scenario for OTel requirement otel-2. */
public final class Otel2WaitResume extends OtelConformanceHandler<String> {

    @Override
    public String handleRequest(Map<String, Object> event, DurableContext context) {
        requireScenario(event, "wait-resume");
        context.wait("otel-wait", Duration.ofSeconds(1));
        return context.step("otel-after-resume", String.class, step -> "resumed");
    }
}
